### PR TITLE
Added algorithm for binary exponent

### PR DESCRIPTION
**Pull request by NEERAJAP2001**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
On my machine I do `bin_exp(0, -2)` and it seems to go into infinite loop mode.
Perhaps you need to raise a ValueError if the exponent is negative?

**Comment by NEERAJAP2001** [comment]
Yes sir, I agree but Travis failed when I used exception handling could you please help me out with that.

**Comment by cclauss** [comment]
```python
if expo < 0:
    raise ValueError(f"expo ({expo}) must not be negative")
```

**Comment by NEERAJAP2001** [comment]
Thank you, sir.

**Comment by NEERAJAP2001** [comment]
@cclauss  sir, should I run falke8, black and then commit again?

File: maths/binary_exponent.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    while expo:
+        div, mod = divmod(expo, 2)
+        if mod:
+            result *= num
+            expo -= 1
+        else:
+            num *= num
+            expo = div
+    return result
+# call the testmod function
```
Comment (cclauss) [inline maths/binary_exponent.py:27]:
```suggestion


```

File: maths/binary_exponent.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    >>> bin_exp(1, 0)
+    1
+    >>> bin_exp(0, -2)
+    'expo must not be negative'
+    >>> bin_exp(-2, -2)
+    'expo must not be negative'
+    """
+    result = 1
+    if expo < 0:
+        return 'expo must not be negative'
```
Comment (cclauss) [inline maths/binary_exponent.py:17]: Line 1 promises that this function will return an int so we can not return a str.  Please raise the ValueError instead.  Search this repo for ValueError to see other algorithms with doctests that deal with exceptions.

Comment (NEERAJAP2001) [reply maths/binary_exponent.py:17]: Oh okay sir, I'll do it.

Comment (cclauss) [reply maths/binary_exponent.py:17]: Nice!  Mypy even agrees with me https://travis-ci.com/github/TheAlgorithms/Python/builds/182944036#L459

**Pull request closed without merging**